framework/src/actions/SetupMeshAction.C

```diff
@@ -152,6 +153,34 @@ SetupMeshAction::setupMesh(MooseMesh * mesh)
 void
 SetupMeshAction::act()
 {
+  // switch non-file meshes to be a file-mesh if using a pre-split mesh configuration.
+  if (_app.parameters().get<bool>("use_split"))
```

Comment:
You shouldn't need to run this logic twice (or more). You only need to run it during the "setup_mesh" task.

Reply:
Not sure what you are saying here.  Could you clarify?

Reply:
Take a look at the code below this if statement (e.g. `if _current_task == ...`). This Action gets fired multiple time to perform different tasks. You only need to run this new logic for the setup task, not the init task.

Reply:
To be clear, it's not about efficiency or anything, just getting it right logically so we don't look back at this later and wonder why this block runs for every task.

Reply:
Ah - I see - good catch.  It is easy to forget that multiple tasks can use/run the same action.